src/platform/vita/vita_keyboard.c

```diff
@@ -0,0 +1,173 @@
+static uint8_t ime_input_text_utf8[SCE_IME_DIALOG_MAX_TEXT_LENGTH + 1];
+
+static void utf16_to_utf8(uint16_t *src, uint8_t *dst)
+{
+    int i;
+    for (i = 0; src[i]; i++) {
+        if ((src[i] & 0xFF80) == 0) {
+            *(dst++) = src[i] & 0xFF;
+        } else if((src[i] & 0xF800) == 0) {
+            *(dst++) = ((src[i] >> 6) & 0xFF) | 0xC0;
+            *(dst++) = (src[i] & 0x3F) | 0x80;
+        } else if((src[i] & 0xFC00) == 0xD800 && (src[i + 1] & 0xFC00) == 0xDC00) {
+            *(dst++) = (((src[i] + 64) >> 8) & 0x3) | 0xF0;
+            *(dst++) = (((src[i] >> 2) + 16) & 0x3F) | 0x80;
+            *(dst++) = ((src[i] >> 4) & 0x30) | 0x80 | ((src[i + 1] << 2) & 0xF);
+            *(dst++) = (src[i + 1] & 0x3F) | 0x80;
+            i += 1;
+        } else {
+            *(dst++) = ((src[i] >> 12) & 0xF) | 0xE0;
+            *(dst++) = ((src[i] >> 6) & 0x3F) | 0x80;
+            *(dst++) = (src[i] & 0x3F) | 0x80;
+        }
+    }
+
+    *dst = '\0';
+}
+
+static void utf8_to_utf16(uint8_t *src, uint16_t *dst)
+{
+    int i;
+    for (i = 0; src[i];) {
+        if ((src[i] & 0xE0) == 0xE0) {
+            *(dst++) = ((src[i] & 0x0F) << 12) | ((src[i + 1] & 0x3F) << 6) | (src[i + 2] & 0x3F);
+            i += 3;
+        } else if ((src[i] & 0xC0) == 0xC0) {
+            *(dst++) = ((src[i] & 0x1F) << 6) | (src[i + 1] & 0x3F);
+            i += 2;
+        } else {
+            *(dst++) = src[i];
+            i += 1;
+        }
+    }
+
+    *dst = '\0';
+}
+
+static int init_ime_dialog(char *title, const char *initial_text, int max_text_length, int type, int option)
+{
+    if (ime_dialog_running)
+    return -1;
+
+    // Convert UTF8 to UTF16
+    memset(ime_title_utf16, 0, sizeof(ime_title_utf16));
+    memset(ime_initial_text_utf16, 0, sizeof(ime_initial_text_utf16));
+    utf8_to_utf16((uint8_t *)title, ime_title_utf16);
+    utf8_to_utf16((uint8_t *)initial_text, ime_initial_text_utf16);
+
+    //clear previous results
+    memset(ime_input_text_utf16, 0, sizeof(ime_input_text_utf16));
+    memset(ime_input_text_utf8, 0, sizeof(ime_input_text_utf8));
+
+    SceImeDialogParam param;
+    sceImeDialogParamInit(&param);
+
+    param.supportedLanguages = 0x0001FFFF;
+    param.languagesForced = SCE_TRUE;
+    param.type = type;
+    param.option = option;
+    if (option == SCE_IME_OPTION_MULTILINE)
+        param.dialogMode = SCE_IME_DIALOG_DIALOG_MODE_WITH_CANCEL;
+    param.title = ime_title_utf16;
+    param.maxTextLength = max_text_length;
+    param.initialText = ime_initial_text_utf16;
+    param.inputTextBuffer = ime_input_text_utf16;
+
+    int res = sceImeDialogInit(&param);
+    if (res >= 0) {
+        ime_dialog_running = 1;
+        ime_dialog_option = option;
+    }
+
+    return res;
+}
+
+static uint8_t *get_ime_dialog_input_text_utf8(void)
+{
+    return ime_input_text_utf8;
+}
+
+static int update_ime_dialog(void) {
+    if (!ime_dialog_running)
+    return IME_DIALOG_RESULT_NONE;
```

Comment:
Here as well.

Reply:
Fixed in 35f42ab00e2b28a49e966a752eb3016a334d53bb